lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1331,6 +1331,39 @@ void SPIRVToLLVM::transFunctionPointerCallArgumentAttributes(
   }
 }
 
+/// When spirv is generated from LLVM IR with opapque pointer enabled and then
+/// the spirv is translated to LLVM IR with typed pointer, function pointer is
+/// casted to i8* type in \p Ty and \p Initializer, which causes error in LLVM
+/// IR verifier. E.g.
+///   [1 x %0][%0 { i32 1, i8* bitcast (void ()* @ctor to i8*), i8* null }]
+/// This function removes the cast so that LLVM IR is valid.
+static void postProcGlobalCtorDtorTypeInit(Type *&Ty, Constant *&Initializer) {
```

Comment:
The function is defined to be void, as if it would cause a side effect rather than convert one statement into another, as it actually does by reassigning an argument `Initializer` . It would require more time to understand the underlying logic comparing to line 1595 style of
` Initializer = dyn_cast<Constant>(transValue(Init, F, BB, false));`.
In my opinion, it's better to return a pointer than to change the `*&` argument.